internal/exp/metrics/staleness/staleness.go

```diff
@@ -0,0 +1,61 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package staleness
+
+import (
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/identity"
+)
+
+// We override how Now() is returned, so we can have deterministic tests
+var nowFunc = time.Now
+
+type Map[T any] interface {
+	// Load the value at key. If it does not exist, the boolean will be false and the value returned will be the zero value
+	Load(key identity.Stream) (T, bool)
+	// Store the given key value pair in the map
+	Store(key identity.Stream, value T)
+	// LoadOrStore will either load the value from the map and return it and the boolean `true`
+	// or if it doesn't exist in the Map yet, the value passed in will be stored and then returned with the boolean `false`
+	LoadOrStore(key identity.Stream, value T) (T, bool)
+	// Remove the value at key from the map
```

Comment:
would like to revert my previous opinion, this is actually a lot less necessary than I thought.
lets remove it to keep the api surface smaller ;)